### ISSUE TITLE
createInput() for file that doesn't exist not returning null in 3.3.1

### ISSUE THREAD

**Issue by linux-man** [processing/processing#5026]
at line 8 of showcasecontrols
`pnlControls = new GPanel(this, pX, pY, pWidth, pHeight, "Panel Tab Text (drag to move : click to open/close)");`
with
```
===================================================
   G4P V4.1.1 created by Peter Lager
===================================================
java.io.IOException: Stream closed
	at java.io.BufferedInputStream.getInIfOpen(BufferedInputStream.java:159)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
	at processing.core.PApplet.loadBytes(PApplet.java:7180)
	at processing.core.PApplet.loadBytes(PApplet.java:7156)
	at processing.core.PApplet.loadImage(PApplet.java:5354)
	at processing.core.PApplet.loadImage(PApplet.java:5278)
	at g4p_controls.GCScheme.makeColorSchemes(Unknown Source)
	at g4p_controls.GAbstractControl.<init>(Unknown Source)
	at g4p_controls.GAbstractControl.<init>(Unknown Source)
	at g4p_controls.GTextBase.<init>(Unknown Source)
	at g4p_controls.GPanel.<init>(Unknown Source)
	at G4P_Showcase.makeDemoControls(G4P_Showcase.java:251)
	at G4P_Showcase.setup(G4P_Showcase.java:60)
	at processing.core.PApplet.handleDraw(PApplet.java:2413)
	at processing.awt.PSurfaceAWT$12.callDraw(PSurfaceAWT.java:1547)
	at processing.core.PSurfaceNone$AnimationThread.run(PSurfaceNone.java:316)
USER DEFINED colour schema active
```
Maybe a G4P bug, but a minor update shouldn't bring incompatibilities of this magnitude. In fact, I didn't try all, but it looks like all G4P demos are broken. Demos run on v3.3.

Linux Mint 8.2 64 bits, Java(TM) SE Runtime Environment (build 1.8.0_131-b11).

**Comment by benfry** [comment]
You'll have to check with the G4P folks… I don't have any way to debug this because I don't know what they're doing in their code. There were some changes to how buffering of input and output streams, but I can't think of anything else that would cause a problem here.

**Comment by benfry** [comment]
Closing for now, but happy to re-open if this is a regression introduced in 3.3.1.

**Comment by quarks** [comment]
The problem is caused by the PApplet method createInput(String). Try this code 
```
InputStream is = createInput("non-existant.png");
System.out.println("IS: " + is);
```
in Processing 3.3 it returns null but in Processing 3.3.1 it returns an InputStream even though the file does not exist.

This means that in 3.3.1 G4P thinks the user has supplied an image to be used for the colour scheme when in fact they haven't.

The createInput method calls createInputRaw which I have tested in both 3.3 and 3.3.1 and in both cases it returned null for a non-existant file. I will modify G4P to use this method instead.

It would be nice to add a Processing method to test that a file exists in the sketch/data folder.

**Comment by benfry** [comment]
Ok, so it may be a Java runtime change because we moved up to 8u131 in 3.3.1. This happened once in the past when they started returning a `sun.plugin.cache.EmptyInputStream` object. Wonder if they changed the type…

**Comment by benfry** [comment]
Agh, nope—just a typo by me.

**Comment by benfry** [comment]
Fixed for 3.3.2 with https://github.com/processing/processing/commit/c0b05cf1c6dfd7b72ae3e37fcbb8805e7bf0fe0b 

Thanks for the report, @linux-man and for looking into it, @quarks.
